bigtable/client/internal/table.cc

```diff
@@ -36,6 +36,9 @@ std::vector<FailedMutation> Table::Apply(SingleRowMutation&& mut) {
   // Build the RPC request, try to minimize copying.
   btproto::MutateRowRequest request;
   request.set_table_name(table_name_);
+  if (!app_profile_id_.empty()) {
```

Comment:
I am seeing a pattern where a proto of some sort is modified based on table name and profile id, consider creating a helper function:

```C++
template <typename Request>
void SetCommonTableOperationRequest(Request& request, std::string table_name, std::string app_profile_id) {
  request.set_table_name(std::move(table_name));
  request.set_app_profile_id(std::move(app_profile_id));
}
```

We may want to create a struct to represent the `table_name, app_profile_id` combo, because they always go together and we want to avoid argument order problems.